Sources/TuistLoader/SwiftPackageManager/PackageInfoMapper.swift

```diff
@@ -397,25 +403,38 @@ public final class PackageInfoMapper: PackageInfoMapping {
         path: AbsolutePath,
         packageFolder: AbsolutePath,
         productTypes: [String: XcodeGraph.Product],
-        productDestinations: [String: XcodeGraph.Destinations],
+        packageSettings: TuistCore.PackageSettings,
         baseSettings: XcodeGraph.Settings,
-        targetSettings: [String: XcodeGraph.SettingsDictionary]
+        targetSettings: [String: XcodeGraph.SettingsDictionary],
+        onlySPMProject: Bool
     ) throws -> ProjectDescription.Target? {
         // Ignores or passes a target based on the `type` and the `packageType`.
         // After that, it assumes that no target is ignored.
         switch target.type {
         case .regular, .system, .macro:
             break
-        case .test, .executable:
+        case .executable:
             switch packageType {
-            case .external:
-                logger.debug("Target \(target.name) of type \(target.type) ignored")
+            case .remote:
+                logger.debug("Target \(target.name) of type \(target.type) is ignored.")
                 return nil
             case .local:
                 break
             }
+        case .test:
+            switch packageType {
+            case .remote:
+                logger.debug("Target \(target.name) of type \(target.type) is ignored.")
+                return nil
+            case .local:
+                if onlySPMProject { break }
+                guard packageSettings.includeLocalPackageTestTargets else {
```

Comment:
```suggestion
                if !packageSettings.includeLocalPackageTestTargets { ... return nil }
```

I think this reads better than using a `guard` here. We're not guarding here, it's more of an `if-else` situation where the `else` block equals to `break` that we don't write explicitly